src/app/components/product-owner/room/room.component.ts

```diff
@@ -9,49 +12,118 @@ import {RoomService} from "../../../services/room.service";
 })
 export class RoomComponent implements OnInit {
 
-  roomId:string;
-  tasks:any[];
-  taskToEstimate:any;
-  estimation: number[];
+  roomId: string;
+  tasks: any[] = [];
+  taskToEstimate: any;
+  estimation: number[] = [];
   estimationMedian: number;
 
-  constructor(private route: ActivatedRoute, private service:RoomService) { }
+  constructor(private route: ActivatedRoute, private service: RoomService, public dialog: MatDialog) {
+  }
 
   ngOnInit() {
     this.roomId = this.route.snapshot.params.id;
-    this.service.getTasks(this.roomId).subscribe(
-      data => this.tasks = data,
-      error => console.error(error)
+    this.service.roomId = this.roomId;
+    this.fetchTasks();
+    this.sendToWebSocket({roomId: this.roomId, type: 'init-host'});
+    this.listenOnWebSockets();
+    this.signUp()
+  }
+
+  private fetchTasks() {
+    this.service.getTasks().subscribe(
+      (data: any) => {
+        this.tasks = data;
+        this.service.tasks = data;
+      }, error => console.error(error)
     )
+  }
 
+  private signUp() {
+    setTimeout(() => {
+      let ref = this.dialog.open(CreateUserComponent);
+      ref.afterClosed().subscribe(
+        data => {
+          if (data) {
+            this.service.setHostUser(data);
+          }
+        }
+      )
+    }, 10);
   }
 
-  selectedToEstimate(task){
-    this.taskToEstimate = task;
-    this.service.selectTaskToEstimate(task).subscribe(
-      ()=> {},
-      error => console.error(error)
+  private listenOnWebSockets() {
+    this.service.websocket.onMessage(
+      (msg: MessageEvent) => {
+        console.log("onMessage ", msg.data);
```

Comment:
Needed or missed?

Reply:
this one is needed, because of lack of the WS documentation 🤦‍♂️